modular_skyrat/modules/cortical_borer/code/cortical_borer.dm

```diff
@@ -0,0 +1,602 @@
+
+//we need a way to produce offspring
+/datum/action/cooldown/produce_offspring
+	name = "Produce Offspring (100 chemicals)"
+	cooldown_time = 1 MINUTES
+	icon_icon = 'modular_skyrat/modules/cortical_borer/icons/actions.dmi'
+	button_icon_state = "reproduce"
+
+/datum/action/cooldown/produce_offspring/Trigger()
+	if(!IsAvailable())
+		to_chat(owner, span_warning("This action is still on cooldown!"))
+		return
+	if(!iscorticalborer(owner))
+		to_chat(owner, span_warning("You must be a cortical borer to use this action!"))
+		return
+	var/mob/living/simple_animal/cortical_borer/cortical_owner = owner
+	if(cortical_owner.host_sugar())
+		to_chat(owner, span_warning("Sugar inhibits your abilities to function!"))
+		return
+	if(!ishuman(cortical_owner.loc))
+		to_chat(cortical_owner, span_warning("You must be inside a human in order to do this!"))
+		return
+	if(cortical_owner.chemical_storage < 100)
+		to_chat(cortical_owner, span_warning("You require at least 100 chemical units before you can reproduce!"))
+		return
+	cortical_owner.chemical_storage -= 100
+	var/list/mob/dead/observer/candidates = pollGhostCandidates("Do you want to spawn as a cortical borer?", ROLE_PAI, FALSE, 100, POLL_IGNORE_CORTICAL_BORER)
+	if(!LAZYLEN(candidates))
+		to_chat(cortical_owner, span_notice("No available borers in the hivemind."))
+		cortical_owner.chemical_storage = min(cortical_owner.max_chemical_storage, cortical_owner.chemical_storage + 100)
+		return
+	var/turf/human_turf = get_turf(cortical_owner.human_host)
+	var/mob/dead/observer/pick_candidate = pick(candidates)
+	var/mob/living/simple_animal/cortical_borer/spawn_borer = new /mob/living/simple_animal/cortical_borer(human_turf)
+	new /obj/effect/decal/cleanable/vomit(human_turf)
+	playsound(human_turf, 'sound/effects/splat.ogg', 50, TRUE)
+	spawn_borer.ckey = pick_candidate.ckey
+	StartCooldown()
+
+/datum/action/cooldown/revive_host
+	name = "Revive Host (200 chemicals)"
+	cooldown_time = 2 MINUTES
+	icon_icon = 'modular_skyrat/modules/cortical_borer/icons/actions.dmi'
+	button_icon_state = "revive"
+
+/datum/action/cooldown/revive_host/Trigger()
+	if(!IsAvailable())
+		to_chat(owner, span_warning("This action is still on cooldown!"))
+		return
+	if(!iscorticalborer(owner))
+		to_chat(owner, span_warning("You must be a cortical borer to use this action!"))
+		return
+	var/mob/living/simple_animal/cortical_borer/cortical_owner = owner
+	if(cortical_owner.host_sugar())
+		to_chat(owner, span_warning("Sugar inhibits your abilities to function!"))
+		return
+	if(!ishuman(cortical_owner.loc))
+		to_chat(cortical_owner, span_warning("You must be inside a human in order to do this!"))
+		return
+	if(cortical_owner.chemical_storage < 200)
+		to_chat(cortical_owner, span_warning("You require at least 100 chemical units before you can reproduce!"))
+		return
+	cortical_owner.chemical_storage -= 200
+	cortical_owner.human_host.revive(full_heal = TRUE) //lets not heal all the way, but do it from the start
```

Comment:
will this restore blood, burn, toxin, clone, traumas, and everything? Seems very strong even if you take 50% brute